importer/tasks.py

```diff
@@ -3,11 +3,55 @@
 """
 from __future__ import unicode_literals
 
+import logging
+
+from django.conf import settings
+
 from lore.celery import async
-from importer.api import import_course_from_file
+from learningresources.api import update_xanalytics
+from xanalytics import send_request, get_result, get_xanalytics_file
+
+log = logging.getLogger(__name__)
 
 
 @async.task
 def import_file(path, repo_id, user_id):
     """Asynchronously import a course."""
+    from importer.api import import_course_from_file
     import_course_from_file(path, repo_id, user_id)
+
+
+@async.task
+def populate_xanalytics_fields(course_id):
+    """
+    Initiate request to xanalytics API to get stats for a course,
+    then trigger async job to retrieve results when they become available.
+    Args:
+        course_id (int): primary key of a Course
+    """
+    if settings.XANALYTICS_URL != "":
+        token = send_request(settings.XANALYTICS_URL + "/create", course_id)
```

Comment:
I'm sure this is probably out of scope if they have already started the API, but why do the RESTful way of doing jobs on the remote end by returning a 202 with a location header on where to check for status, and then the status page on the remote end returns a 200 until it is done ala http://restcookbook.com/Resources/asynchroneous-operations/ or https://www.adayinthelifeof.nl/2011/06/02/asynchronous-operations-in-rest/